packs/minecraft_java_edition/variables.hcl

```diff
@@ -0,0 +1,541 @@
+variable "priority" {
+  type        = number
+  description = "Priority for the Job."
+  default     = 99
+}
+
+# see https://developer.hashicorp.com/nomad/docs/concepts/architecture#regions
+variable "region" {
+  type        = string
+  description = "Region for the Job."
+  default     = "global"
+}
+
+# see https://developer.hashicorp.com/nomad/docs/job-specification/resources
+variable "resources" {
+  type = object({
+    cpu        = number
+    cores      = number
+    memory     = number
+    memory_max = number
+  })
+
+  description = "Resource Limits for the Application."
+
+  default = {
+    # Tasks can ask for `cpu` or `cores`, not both.
+    # value in MHz
+    cpu = 4000
+
+    # Tasks can ask for `cpu` or `cores`, not both.
+    # see https://developer.hashicorp.com/nomad/docs/job-specification/resources#cores
+    # and https://developer.hashicorp.com/nomad/docs/drivers/docker#cpu
+    cores = null
+
+    # value in MB
+    # 2048 = ~10 players, 4096 = ~25 players
+    # 6144 = ~40 players, 8192 = ~90 players
+    # 10240 = ~150+ players
+    # see https://apexminecrafthosting.com/how-much-ram-do-i-need-for-my-server/
+    memory = 4096
+
+    # value in MB
+    # see https://developer.hashicorp.com/nomad/docs/job-specification/resources#memory-oversubscription
+    # and https://developer.hashicorp.com/nomad/docs/drivers/docker#memory
+    memory_max = 5120
+  }
+}
+
+variable "service_name_prefix" {
+  type        = string
+  description = "Name for the Service."
+  default     = "minecraft"
+}
+
+variable "service_provider" {
+  type        = string
+  description = "Provider for the Service."
+  default     = "consul"
+}
+
+variable "restart_logic" {
+  type = object({
+    attempts = number
+    interval = string
+    delay    = string
+    mode     = string
+  })
+
+  description = "Restart Logic for the Application."
+
+  default = {
+    attempts = 3
+    interval = "120s"
+    delay    = "30s"
+    mode     = "fail"
+  }
+}
+
+variable "task_name" {
+  type        = string
+  description = "Name for the Task."
+  default     = "minecraft"
+}
+
+variable "verbose_output" {
+  type        = bool
+  description = "Toggle to enable verbose output."
+  default     = true
+}
+
+variable "volumes" {
+  type = map(object({
+    name        = string
+    type        = string
+    destination = string
+    read_only   = bool
+  }))
+
+  description = "Volumes for the Application."
+
+  default = {
+    minecraft_data = {
+      name        = "minecraft_data",
+      type        = "host"
+      destination = "/data"
+      read_only   = false
+    },
+
+    minecraft_extras = {
+      name        = "minecraft_extras",
+      type        = "host"
+      destination = "/extras"
+      read_only   = false
+    },
+
+    minecraft_worlds = {
+      name        = "minecraft_worlds",
+      type        = "host"
+      destination = "/worlds"
+      read_only   = false
+    },
+  }
```

Comment:
There shouldn't be a default here IMO, because if those `host_volumes` don't exist the deployment will fail.